[JOJI-727] 사용자 드롭다운 메뉴 구현

MuteAim opened the pull request:
- 가상의 로그인 상태를 만들어서 로그인 여부에 따른 네비게이션 상태 변화

- 로그인시
![image](https://github.com/user-attachments/assets/dea34a43-7e89-427d-84ef-10514041fa1d)


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **새로운 기능**
	- 사용자 인증 상태 관리를 위한 네비게이션 바 업데이트
	- 로그인/로그아웃 기능 추가
	- 메뉴 구조 개선 (FAQ, 구독, 문의 메뉴 추가)
	- 공지사항 카테고리로 제한된 게시물 가져오기 기능 추가
	- 개발 서버 포트 설정 변경 (포트 80으로 설정)

- **버그 수정**
	- 결제 페이지 링크 오타 수정
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

MuteAim committed:
feat: 네비게이션 바 업데이트

- 가상의 로그인 상태를 만들어서
로그인 여부에 따른 네비게이션 상태 변화

MuteAim committed:
feat: easilyPosts 수정

- 취지에 맞게 NOTICE만 보여주도록 변경

MuteAim committed:
feat: 드롭다운 수정

-메뉴와 사용자 드롭다운을 분리하여 동시에 드롭다운이 되는 현상 개선

AlangGY commented on src/app/_components/navBar.js:
```diff
@@ -92,18 +103,52 @@ export default function NavBar() {
-        {/* Right: Dashboard and Login Buttons */}
         <div className="flex items-center gap-4">
-          <Link href="/dashboard">
+          <Link href="https://easily-dashboard.jojicompany.com/dashboard/proposal/create">
```
그냥 root로 이동시켜주는게 좋을것 같아욥.

`https://easily-dashboard.jojicompany.com`

MuteAim committed:
feat: 로그인상태에 따른 네비게이션 바(로그인 처리)

- 쿠키에 엑세스토큰이 존재할경우 Welcome, 존재하지 않을경우에는 로그인버튼이 표시됨

AlangGY commented on package.json:
```diff
@@ -7,7 +7,7 @@
   "engineStrict": true,
   "scripts": {
-    "dev": "next dev --turbopack",
+    "dev": "next dev --turbopack --port=80",
```
오호 저건 새로 안 사실이네요. 뭐, 신경쓰실필욘 없어보입니다.

AlangGY commented on src/app/_components/navBar.js:
```diff
@@ -1,10 +1,47 @@
+  const [isLoggedIn, setIsLoggedIn] = useState(false); // 로그인 상태 관리
+  const [userName, setUserName] = useState(""); // 사용자 이름
+
+  useEffect(() => {
+    fetch(`${process.env.NEXT_PUBLIC_BASE_URL}/auth/profile`, {
+      method: "GET",
+      credentials: "include",
+    })
+      .then((response) => {
+        if (response.status === 200) {
+          return response.json();
+        } else if (response.status === 401) {
+          throw new Error("Unauthorized");
+        } else {
+          throw new Error("Unexpected error");
+        }
+      })
+      .then((data) => {
+        setIsLoggedIn(true);
+        setUserName(data.name);
+      })
+      .catch((error) => {
+        if (error.message === "Unauthorized") {
+          setIsLoggedIn(false);
+        }
+      });
+  }, []);
```
로그인 로딩 상태 반영되면 좋겠네요!
인증 진행과정동안 로그인 버튼이 표시되고 있는 상황은 그리 좋아 보이진 않으니까요

AlangGY reviewed (changes requested):
크으 마법같다 마법!

조금 더 가다듬으면 좋겠는부분들을 코멘트로 남겨두었습니다 ㅎ

MuteAim committed:
feat: 로그인 상태 개선

현재 세가지의 상태가 존재합니다.
1. 로그인이 되어있는 상태 -> Welcome
2. 로그인 시도중인 상태 ->
아무것도 나오지 않음
3. 로그인이 안되어있는 상태 ->
로그인 버튼 표시

MuteAim committed:
feat: easilyPosts 개선

AlangGY reviewed (approved):
작업한지 일주일이 지나서 그런지, 반영하고 보는게 젤 좋을것 같네요. 이대로 머지해주세요!
